packages/@aws-cdk/aws-codecommit/lib/pipeline-action.ts

```diff
@@ -54,11 +54,15 @@ export class PipelineSourceAction extends codepipeline.SourceAction {
       configuration: {
         RepositoryName: props.repository.repositoryName,
         BranchName: props.branch || 'master',
-        PollForSourceChanges: props.pollForSourceChanges !== undefined ? props.pollForSourceChanges : true
+        PollForSourceChanges: props.pollForSourceChanges || false,
       },
       outputArtifactName: props.outputArtifactName
     });
 
+    if (props.pollForSourceChanges !== true) {
```

Comment:
Ideally you wouldn't want to codify the default multiple times in your code.

Since the default is false, I would just write this `if (!props.pollForSourceChanges)`.
Alternatively, if you want to be more formal, create a local variable:

```ts
const pollForSourceChanges = props.pollForSourceChanges || false;
```

And use it everywhere.

Reply:
We can't use a variable here, as the call to `super` must be the first statement in the constructor.

Will change to `if (!props.pollForSourceChanges)`.

Reply:
That’s not correct. You can definitely use a variable :-)

Reply:
Can you write out the code for that? Cause I can't see it.